runtime/src/main/cpp/Memory.cpp

```diff
@@ -956,6 +1054,7 @@ ALWAYS_INLINE void runDeallocationHooks(ContainerHeader* container) {
       cyclicRemoveAtomicRoot(obj);
     }
 #endif  // USE_CYCLIC_GC
+    CycleDetector::instance().removeCandidateIfNeeded(obj);
```

Comment:
That's a mistake. The instance should not be created at all if leak checker is disabled. Will fix.